ros/src/driveworks/packages/autoware_driveworks_gmsl_interface/CMakeLists.txt

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) 2016, NVIDIA CORPORATION.  All rights reserved.
+
+cmake_minimum_required(VERSION 2.8.3)
+project(gmsl_camera)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error -Wno-deprecated-declarations -std=c++11")
+
+##################
+## ROS packages ##
+##################
+## Find catkin macros and libraries
+## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
+## is used, also find other catkin packages
+find_package(catkin REQUIRED COMPONENTS
+  roscpp
+  sensor_msgs
+  std_msgs
+  cv_bridge
+  image_transport
+  camera_info_manager
+  nodelet
+)
+
+
+##############################
+## Set basic configurations ##
+##############################
+set(CMAKE_MODULE_PATH /usr/local/driveworks/samples/cmake)
+set(Driveworks_INCLUDE_DIR /usr/local/driveworks/include)
+set(Driveworks_LIBRARY /usr/local/driveworks/lib)
+
+link_directories(${Driveworks_LIBRARY})
+
+
+#####################
+## System packages ##
+#####################
+find_package(CUDA REQUIRED)
```

Comment:
Given that users may not have the CUDA and Driveworks libraries installed, this package should use the following structure:

```
find_package(CUDA)
if(CUDA_FOUND)
# The rest of the CMake code goes here
...
endif()
```

See for example: https://github.com/CPFL/Autoware/blob/develop/ros/src/computing/perception/localization/lib/ndt_gpu/CMakeLists.txt#L21